shaky/src/main/java/com/linkedin/android/shaky/Shaky.java

```diff
@@ -186,12 +188,23 @@ public void hearShake() {
         arguments.putInt(ShakySettingDialog.SHAKY_CURRENT_SENSITIVITY, delegate.getSensitivityLevel());
         SendFeedbackDialog sendFeedbackDialog = new SendFeedbackDialog();
         sendFeedbackDialog.setArguments(arguments);
-        sendFeedbackDialog.show(activity.getFragmentManager(), SEND_FEEDBACK_TAG);
+        if (customFragment != null) {
+            customFragment.show(activity.getFragmentManager(), CUSTOM_DIALOG_TAG);
+        } else {
+            sendFeedbackDialog.show(activity.getFragmentManager(), SEND_FEEDBACK_TAG);
+        }
         if (shakyFlowCallback != null) {
             shakyFlowCallback.onUserPromptShown();
         }
     }
 
+    /**
+     * Custom dialog to be shown before the feedback flow
+     */
+    public void setCustomDialog(@NonNull DialogFragment fragment) {
+        customFragment = fragment;
```

Comment:
wouldn't it make more sense to add this field to the `ShakeDelegate`? It's giving me the vibe that communicating little customizations like this is the purpose of the delegate.

Reply:
So if you look at the current flow altogether, you can only alter the dialog title and message and not the button text or behavior. And also, the delegate starts the feedback flow whereas we are adding a feature that could be run before the feedback flow if a custom dialog is provided